lbry/db/database.py

```diff
@@ -277,6 +277,17 @@ async def get_purchases(self, **constraints) -> Result[Output]:
         return await self.fetch_result(q.get_purchases, **constraints)
 
     async def search_claims(self, **constraints) -> Result[Output]:
+        if 'channel' in constraints:
+            channel_url = constraints.pop('channel')
+            match = await self.resolve([channel_url])
+            if isinstance(match, dict):
+                for value in match.values():
+                    if isinstance(value, Output):
+                        constraints['channel_hash'] = value.claim_hash
+                    else:
+                        return Result([], 0)
+            else:
+                return Result([], 0)
```

Comment:
It would be a lot more performant to do this in `q.search_claims` because that cuts in half the main process <-> sub tasks communication. I ended up implementing this in my branch earlier: https://github.com/lbryio/lbry-sdk/blob/cae800565dc276c4b05cd93738aa82e757596b07/lbry/db/queries/search.py#L284-L291